libs/librepcb/common/dialogs/aboutdialog.cpp

```diff
@@ -0,0 +1,46 @@
+#include "aboutdialog.h"
+#include "ui_aboutdialog.h"
+
+AboutDialog::AboutDialog(QWidget* parent, const librepcb::Version& appVersion, const QString& gitVersion, const QString& buildDate) :
+    QDialog(parent),
+    ui(new Ui::AboutDialog)
+{
+    // Set up base dialog
+    ui->setupUi(this);
+
+    // Content alignment
+    ui->horizontalLayout->setAlignment(ui->logo, Qt::AlignTop);
+    ui->aboutContentLayout->setAlignment(Qt::AlignTop);
+
+    // Add content to about dialog
+    QLabel *textIntro = new QLabel(tr("LibrePCB is a free & open source schematic/layout-editor."));
+    QLabel *headerVersion = new QLabel("<b>" + tr("Version Info") + "</b>");
+    QLabel *textVersion = new QLabel(QString("%1 (%2), built on %3.").arg(appVersion.toPrettyStr(3), gitVersion, buildDate));
+    QLabel *headerLinks = new QLabel("<b>" + tr("Links") + "</b>");
+    QLabel *textLinks = new QLabel(tr("For more information, please check out <a href='%1'>librepcb.org</a><br>or our <a href='%2'>GitHub repository</a>.").arg("http://librepcb.org/", "https://github.com/LibrePCB/LibrePCB"));
+    QLabel *headerLicense = new QLabel("<b>" + tr("License") + "</b>");
+    QLabel *textLicense = new QLabel(tr("LibrePCB is free software, released under the GNU General<br>Public License (GPL) version 3 or later. You can find the full<br>license text <a href='https://github.com/LibrePCB/LibrePCB/blob/master/LICENSE.txt'>in our source code</a>."));
+    headerVersion->setContentsMargins(0, 16, 0, 0);
+    headerLinks->setContentsMargins(0, 16, 0, 0);
+    headerLicense->setContentsMargins(0, 16, 0, 0);
+    textLinks->setOpenExternalLinks(true);
+    textLicense->setOpenExternalLinks(true);
+    ui->aboutContentLayout->addWidget(textIntro);
+    ui->aboutContentLayout->addWidget(headerVersion);
+    ui->aboutContentLayout->addWidget(textVersion);
+    ui->aboutContentLayout->addWidget(headerLinks);
+    ui->aboutContentLayout->addWidget(textLinks);
+    ui->aboutContentLayout->addWidget(headerLicense);
+    ui->aboutContentLayout->addWidget(textLicense);
```

Comment:
Why creating all widgets by code instead of using the designer?

Reply:
Because of the dynamic content in the strings. How would I do that with the designer?

Reply:
All the widgets can be created and configured in the designer. Then set reasonable widget names (for example `lblLicenseText`). Dynamic texts can then be set in code like that:

    ui->lblLicenseText->setText(...);

Of course all static texts (like your `textIntro`) should be filled directly in the designer, no need to do this by code.

Reply:
If you look at the current version, I used two helper methods to style headings and text. Is there some way of adding "styles" to widgets to set parameters like `contentsMargins` or `openExternalLinks` without repeating yourself?